posthog/hogql_queries/web_analytics/top_pages.py

```diff
@@ -0,0 +1,151 @@
+from django.utils.timezone import datetime
+
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_select
+from posthog.hogql.query import execute_hogql_query
+from posthog.hogql_queries.web_analytics.web_analytics_query_runner import WebAnalyticsQueryRunner
+from posthog.hogql_queries.utils.query_date_range import QueryDateRange
+from posthog.models.filters.mixins.utils import cached_property
+from posthog.schema import WebTopPagesQuery, WebTopPagesQueryResponse
+
+
+class TopPagesQueryRunner(WebAnalyticsQueryRunner):
```

Comment:
Something with this query is not working for me. I'm getting a `Illegal type String of argument for aggregate function avg.` error. A quick check locally under data management shows that I don't have the `$prev_pageview_max_scroll_percentage` property at all, so it's likely treated as a string, not an integer. I wouldn't consider this blocking for an alpha.

Reply:
Yes, I have the same problem, I'll need to figure this out before we roll it out to new people. I naively added a `toFloat` around this, which didn't work either, but I'm sure there must be something I can do.

Reply:
Possibly a very naive toString -> toFloat might work?
The alternative is to make sure that the required property definitions exist with the right types, when loading the page.